app/helpers/quadicon_helper.rb

```diff
@@ -1,45 +1,180 @@
 module QuadiconHelper
-  # render a quadicon
-  #   size: size of the quadicon
-  #   item: element to be represented
-  #   typ:  type of view (:listnav, ....?) # FIXME: find possible values
-  #   db:   name of the model of item
-  #   row:
-  #   height:
-  #   mode:
-  def render_quadicon(item, options)
+  # Refactoring phase 1
+  # * Add tests
+  # * Move configuration up and out
+  # * Try to reveal more intention in conditionals
+  # * Extract smaller methods from large methods
+
+  # Collect and normalize global, environment state
+
+  # @settings and `settings` method
+  # @listicon
+  # @embedded
+  # @showlinks
+  # @policy_sim
+  # @explorer
+  # @view.db
+  # @parent
+  # @lastaction
+  # session[:policies]
+  # request.parameters[:controller]
+
+  def quadicon_truncate_mode
+    @settings.fetch_path(:display, :quad_truncate) || 'm'
+  end
+
+  def listicon_nil?
+    @listicon.nil?
+  end
+
+  def quadicon_vm_attributes(item)
+    @quad_vm_attrs ||= vm_quad_link_attributes(item)
+  end
+
+  def quadicon_vm_attributes_present?(item)
+    quadicon_vm_attributes(item) && !quadicon_vm_attributes(item).empty?
+  end
+
+  def quadicon_in_embedded_view?
+    !!@embedded
+  end
+
+  def quadicon_show_link_ivar?
+    !!@showlinks
+  end
+
+  def quadicon_hide_links?
+    quadicon_in_embedded_view? && !quadicon_show_link_ivar?
```

Comment:
I think was mimicking something in the current code, but that's better and should be the same in every case.